server/controller/comments/getAllCommentForAPost.go

```diff
@@ -0,0 +1,45 @@
+package comments
+
+import (
+	"campusburn-backend/dbConnection"
+	"campusburn-backend/model"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+/*
+FLOW TO GET ALL THE COMMENTS FOR A POST:
+---------------------------------------
+1. Get the post Id from the request.
+2. Find all the comments for this post from the database.
+3. If any error while finding the comments, then return error response.
+4. Return the success response with all comments as the data.
+
+*/
+
+func GetAllCommentForAPost(c *fiber.Ctx) error {
+
+	var post []model.Post
+	var postId = c.Params("postId")
+
+	postSearchError := dbConnection.DB.Where("ID = ?", postId).First(&post).Error
+	if postSearchError != nil {
+		return c.Status(fiber.StatusBadRequest).JSON(fiber.Map{
+			"Error": "Post not found",
+		})
+	}
+
+	var allComments []string
+
+	commentSearchError := dbConnection.DB.Model(&model.Comment{}).Where("post_id = ?", postId).Pluck("comment_content", &allComments).Error
+	if commentSearchError != nil {
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"Error": "Error while fetching the comments for the post",
+		})
+	}
+
+	return c.Status(fiber.StatusOK).JSON(fiber.Map{
+		"Message": "Comments fetched successfully",
+		"data":    allComments,
```

Comment:
We don't need to return the whole response array since this would be consisting of some sensitive and personal info like authToken details, password, etc. So, you have to create a type ( struct in golang ). This type should consist of the commentContent, ID of the comment, createdAt, postId, userId, username of the user who created this, etc., and save the response in a new array from the `allComments` array. Then, you have to return this final array.

Reply:
> We don't need to return the whole response array since this would consist of some sensitive and personal info like authToken details, password, etc. So, you have to create a type ( struct in golang ). This type should consist of the comment content, ID of the comment, createdAt, postId, userId, username of the user who created this, etc., and save the response in a new array from the `allComments` array. Then, you have to return this final array.

Ok, I will return that way, in the above I only return comments, not any other stuff.
So, exactly are comment_content, commentId,createdAt,postId,userId, username, like and dislike of the comment these particular things only or need to be added something?

Reply:
> > We don't need to return the whole response array since this would consist of some sensitive and personal info like authToken details, password, etc. So, you have to create a type ( struct in golang ). This type should consist of the comment content, ID of the comment, createdAt, postId, userId, username of the user who created this, etc., and save the response in a new array from the `allComments` array. Then, you have to return this final array.
> 
> Ok, I will return that way, in the above I only return comments, not any other stuff. So, exactly are `comment_content, commentId, createdAt, postId, userId, username, like and dislike of the comment `these particular things only or need to be added something?

The required data acc to me would be:
. CommentContent
. CommentId
. createdAt ( date for comment creation )
. PostId
. UserId ( ID of the user who created this comment )
. Username ( username of the user who created this comment )
. ProfilePhoto of the user
. Email of the user
. LikeCount of that comment
. DislikeCount of that comment


For now, Ig this would be fine, you can log the whole response and see if any other thing is required apart from the above ones.